src/arrayMethodSort.js

```diff
@@ -4,8 +4,30 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (a.toString() > b.toString()) {
+        return 1;
+      } else if (a.toString() < b.toString()) {
+        return -1;
+      } else {
+        return 0;
+      }
+    },
+  ) {
+    let tempValue;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let k = 0; k < this.length - 1; k++) {
+        if (compareFunction(this[k], this[k + 1]) > 0) {
+          tempValue = this[k];
+          this[k] = this[k + 1];
+          this[k + 1] = tempValue;
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm implemented here is correct, but it's not optimized. It will run through the entire array even if the array becomes sorted before the last iteration. Implementing a flag to check if any swaps have been made in the inner loop could optimize the sorting process by exiting early if the array is already sorted.